hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2678,3 +2679,115 @@ def hpat_pandas_series_fillna_impl(self, value=None, method=None, axis=None, inp
+            '{} The function unsupport limit is not None.'.format(_func_name))
+
+    if not isinstance(freq, (types.Omitted, types.NoneType)):
+        raise TypingError(
+            '{} The function unsupport freq is not None.'.format(_func_name))
+
+    def hpat_pandas_series_pct_change_impl(self, periods=1, fill_method='pad', limit=None, freq=None):
+        if not (fill_method is None or fill_method in ['pad', 'ffill', 'backfill', 'bfill']):
+            raise ValueError(
+                "Method pct_change(). Unsupported parameter. The function uses fill_method pad (ffill) or backfill (bfill) or None.")
+        local_series = self.copy()
+        if fill_method is not None:
+            # replacement method fillna for given method
+            # =========================================
+            # Example:
+            # s = [1.1, 0.3, np.nan, 1, np.inf, 0, 1.1, np.nan, 2.2, np.inf, 2, 2]
+            # result = [1.1, 0.3, 0.3, 1, inf, 0, 1.1, 1.1, 2.2, inf, 2, 2]
+            # ==========================================
+            for i in range(len(local_series._data)):
+                # check each element on numpy.nan
+                if numpy.isnan(local_series._data[i]):
+                    if fill_method in ['pad', 'ffill']:
+                        # if it first element is nan, element will be is nan
+                        # if it not first element, element will be is nearest is not nan element
+                        # take a step back while will not find is not nan element
+                        # if before the first element you did not find one, the element will be equal nan
+                        if i == 0:
+                            local_series._data[i] = numpy.nan
+                        else:
+                            k = 1
+                            while numpy.isnan(local_series._data[i - k]):
+                                if i - k == 0:
+                                    local_series._data[i] = numpy.nan
+                                    break
+                                k += 1
+                            local_series._data[i] = local_series._data[i - k]
+                    elif fill_method in ['backfill', 'bfill']:
+                        # if it last element is nan, element will be is nan
+                        # if it not last element, element will be is nearest is not nan element
+                        # take a step front while will not find is not nan element
+                        # if before the last element you did not find one, the element will be equal nan
+                        if i == len(local_series._data)-1:
+                            local_series._data[i] = numpy.nan
+                        else:
+                            k = 1
+                            while numpy.isnan(local_series._data[i + k]):
+                                if i + k == len(local_series._data) - 1:
+                                    local_series._data[i] = numpy.nan
```

Comment:
`local_series._data[i`] is already `nan` in this step, so you can remove the assignment `local_series._data[i] = numpy.nan`.

Reply:
Please consider the comment.

Reply:
> `local_series._data[i`] is already `nan` in this step, so you can remove the assignment `local_series._data[i] = numpy.nan`.

Please consider the comment.